packages/plugins/EVM/src/state/Connection/providers/Opera.ts

```diff
@@ -0,0 +1,25 @@
+import type { ChainId } from '@masknet/web3-shared-evm'
+import { injectedOperaProvider } from '@masknet/injected-script'
+import { ProviderType } from '@masknet/web3-shared-evm'
+import type { EVM_Provider } from '../types'
+import { BaseInjectedProvider } from './BaseInjected'
+import { SharedContextSettings } from '../../../settings'
+
+export class OperaProvider extends BaseInjectedProvider implements EVM_Provider {
+    constructor() {
+        super(ProviderType.Opera, injectedOperaProvider)
+    }
+
+    override get readyPromise() {
+        const isOpera = window.navigator.userAgent.includes('OPR/')
+        if (!isOpera) {
+            return Promise.reject(new Error('Not available on this browser.'))
+        }
+        return this.bridge.untilAvailable().then(() => undefined)
+    }
+    override async switchChain(chainId?: ChainId) {
+        await SharedContextSettings.value.updateAccount({
```

Comment:
The shared context serves only for MaskWallet. For Opera, I think you can do:

1. Check the given `chainId` is supported by Opera. If not, throw a deny error.
2. If the expected `chainId` is not the same as the current one. Reconnect it `this.connect(chainId)`.

Reply:
I tried to do it this way. But unfortunately it didn't work. Changed the chain in Maskbook, but did not change the chain in Opera Crypto Wallet. This change I made works surprisingly flawlessly. Does it need to be changed?